### PR TITLE
Task model controller views

### PR DESCRIPTION
**Pull request by hannako**
- adds some seed data
- creates task model
- creates basic tasks controller. Helper methods eg `get_tasks_for_user` and groups eg `@datasetsBroken` not yet implemented, just pulling in all datasets/tasks
- creates task views. Filters not yet implemented, just hard coded
- imports show-hide js to a component

### CODE REVIEW

File: db/migrate/20170605140131_create_tasks.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class CreateTasks < ActiveRecord::Migration[5.1]
+  def change
+    create_table :tasks do |t|
+      t.references :organisation, foreign_key: true
+      t.integer :quantity, default: 0
+      t.string :required_permission_name
+      t.string :description, limit: 128, default: "", null: true
+      t.string :category, limit: 20
+  #TODO rename to short_name?
```
Comment (govuklaurence) [inline db/migrate/20170605140131_create_tasks.rb]: Absolutely do

File: app/controllers/tasks_controller.rb [hunk 2]
```diff
@@ -1,4 +1,33 @@
...
+    @organisation = current_publishing_user.primary_organisation
+    @datasets = Dataset.all
+    @tasks = get_tasks_for_organisation(@organisation.name)
+    @datasetsUpdate = @datasets.all
+    @datasetsBroken = @datasets.all
+  end
+
+private
+
+  def get_tasks_for_user(user)
```
Comment (govuklaurence) [inline app/controllers/tasks_controller.rb]: Just fix these two CC issues then we can merge

File: config/routes.rb [hunk 3]
```diff
@@ -7,11 +7,11 @@
       sessions: 'publishing_users/sessions'
     }
 
-  root to: 'tasks#index'
+  root to: 'tasks#my'
```
Comment (govuklaurence) [inline config/routes.rb]: This may not merge happily, suggest that you're rebased to the latest master

File: spec/models/task_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Task, type: :model do
```
Comment (govuklaurence) [inline spec/models/task_spec.rb]: Just get rid of this file - if it's not doing anything we don't need it. Integration tests are more useful at this stage.

File: config/routes.rb (hunk 3 above)
Comment (rossjones) [inline config/routes.rb]: Maybe we should route tasks to /tasks, we can redirect there from home if user is logged in.

Comment (govuklaurence) [inline config/routes.rb]: There's routes for that below, and agree, on home controller we should redirect as a before action

**Pull request merged**